tests/Aspire.Hosting.Tests/WithReferenceTests.cs

```diff
@@ -246,6 +246,52 @@ public async Task ConnectionStringResourceWithConnectionString()
         Assert.Contains(config, kvp => kvp.Key == "ConnectionStrings__resource" && kvp.Value == "123");
     }
 
+    [Fact]
+    public async Task ConnectionStringResourceWithExpressiononnectionString()
+    {
+        using var builder = TestDistributedApplicationBuilder.Create();
+
+        var endpoint = builder.AddParameter("endpoint", "http://localhost:3452");
+        var key = builder.AddParameter("key", "secretKey", secret: true);
+
+        var cs = ReferenceExpression.Create($"Endpoint={endpoint};Key={key}");
+        
+        // Get the service provider.
+        var resource = builder.AddConnectionString("cs", cs);
+
+        var projectB = builder.AddProject<ProjectB>("projectb")
+                              .WithReference(resource);
+
+        // Call environment variable callbacks.
+        var config = await EnvironmentVariableEvaluator.GetEnvironmentVariablesAsync(projectB.Resource, DistributedApplicationOperation.Run, TestServiceProvider.Instance).DefaultTimeout();
+
+        var servicesKeysCount = config.Keys.Count(k => k.StartsWith("ConnectionStrings__"));
+        Assert.Equal(1, servicesKeysCount);
+        Assert.Equal("Endpoint=http://localhost:3452;Key=secretKey", config["ConnectionStrings__cs"]);
+    }
+
+    [Fact]
```

Comment:
Add a test like we have in the ParameterEndToEnd playground where we use a resource (such as a SQL DB) as an input to the expression.